### PR TITLE
refactor: convert d2 to app runtime

### PR DESCRIPTION
**Pull request by edoardo**
Implements [TECH-460](https://jira.dhis2.org/browse/TECH-460)

**Relates to https://github.com/dhis2/data-visualizer-app/pull/1369**

---

### Description

Part of the effort to phase out `d2` in favour of `app-runtime`.
All API interaction is being rewritten to use `app-runtime` and move the query use closer to the component that uses it (wherever the query and fetch function does not need to be exported).

---

### BREAKING CHANGE

Apps need to pass the `dataEngine` in place of `d2`

---

### Known issues

-   [ ] `OrgUnitTree` needs to be rewritten to not use `d2.models` in order to use the converted query functions.

### CODE REVIEW

File: src/components/DynamicDimension/DynamicDimension.js [hunk 1]
```diff
@@ -1,29 +1,48 @@
...
-import { apiFetchItemsByDimension } from '../../api/dimensions'
+
+const itemsByDimensionQuery = {
+    itemsByDimension: {
+        resource: 'dimensions',
+        id: ({ id }) => `${id}/items`, // TODO replace with the new way when available
+        params: {
+            fields: 'id,displayName~rename(name)',
+            order: 'displayName:asc',
+            //paging: false, // XXX needed?!
```
Comment (jenniferarnesen) [inline src/components/DynamicDimension/DynamicDimension.js:14]: needs addressing?

File: src/api/organisationUnits-dataEngine.js [hunk 2]
```diff
@@ -0,0 +1,115 @@
+import { onError } from './index'
```
Comment (jenniferarnesen) [inline src/api/organisationUnits-dataEngine.js:1]: Is the goal to remove the old `api/organisationUnits` as soon as Dashboards is converted and OrgUnitDimension is redone with these functions?

Comment (edoardo) [reply src/api/organisationUnits-dataEngine.js:1]: Actually this one with the `dataEngine` postfix will be renamed to `organisationUnits` and replace the old one.
For now we need both.
The reason is that `OrgUnitDimension` uses `d2-ui` components and d2 models are required there.
The new OrgUnitTree component in `ui` is still missing some features.

**Comment by janhenrikoverland** [comment]
Open DV, data, event data items, program Antenatal care, select an item, update -> produces this request:

```dimension=dx:lxAQ7Zs9VYR.sWoqcoByYmD,pe:LAST_12_MONTHS```

Should be:

```
dimension=dx:lxAQ7Zs9VYR.sWoqcoByYmD
dimension=pe:LAST_12_MONTHS
```

**Comment by janhenrikoverland** [comment]
Same as above for program indicators.

**Comment by janhenrikoverland** [comment]
DV, data elements, details (operands), select a group, then enter a filter text -> the request has "{$filterText}" rather than the actual text:

```filter=displayName:ilike:{$filterText}```

**Comment by edoardo** [comment]
> DV, data elements, details (operands), select a group, then enter a filter text -> the request has "{$filterText}" rather than the actual text:
> 
> `filter=displayName:ilike:{$filterText}`

Bug fixed.

File: src/components/DynamicDimension/DynamicDimension.js (hunk 1 above)
Comment (janhenrikoverland) [reply src/components/DynamicDimension/DynamicDimension.js:14]: Could be fixed as part of https://jira.dhis2.org/browse/DHIS2-9274

**Review by janhenrikoverland — APPROVED** [review]
LG, let's get this merged :+1:

**Pull request merged**